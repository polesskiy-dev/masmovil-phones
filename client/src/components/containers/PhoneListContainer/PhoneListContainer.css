.phones-list {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(400px, 1fr));

    grid-column-gap: 32px;
    grid-row-gap: 24px;

    margin: 24px 24px;
}



@media only screen and (max-width: 480px) {
    .phones-list {
        grid-template-columns: repeat(auto-fit, minmax(256px, 1fr));
        margin: 8px 8px;
    }
}
