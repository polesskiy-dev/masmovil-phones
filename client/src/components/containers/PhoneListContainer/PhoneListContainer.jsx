import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';

import getPhones from '../../../selectors/phones.selector';
import PhoneCardComponent from '../../presentional/PhoneCardComponent/PhoneCardComponent';
import withFetchingSpinner from '../../hocs/withFetchingSpinner';
import withPhonesSubscription from '../../hocs/withPhonesSubscribtion';

class PhoneListContainer extends PureComponent {
  static propTypes = {
    phones: PropTypes.array.isRequired,
  };

  render() {
    const { phones } = this.props;

    return <section className="phones-list__wrapper">
      {phones.map(phone => <PhoneCardComponent {...phone} key={phone.DeviceName}/>)}
    </section>;
  }
}

export default compose(
  withPhonesSubscription,
  withFetchingSpinner,
  connect(getPhones)
)(PhoneListContainer);
