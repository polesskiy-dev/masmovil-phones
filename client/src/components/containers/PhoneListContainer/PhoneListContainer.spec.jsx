import React from 'react';
import renderer from 'react-test-renderer';

import withDefaultIntl from '../../../helpers/with-default-intl';
import { PhoneListContainerRaw } from './PhoneListContainer';

jest.mock('../PhoneDetailModal/PhoneDetailModal', () => () => (
  <div>mocked</div>
));

const stubPhones = [
  {
    DeviceName: 'Apple iPhone',
    Brand: 'Apple',
    img:
      'https://upload.wikimedia.org/wikipedia/commons/0/02/IPhone_2G_PSD_Mock.png',
    technology: 'GSM',
    announced: '2007, January. Released 2007, June',
    dimensions: '115 x 61 x 11.6 mm (4.53 x 2.40 x 0.46 in)',
    type: 'TFT capacitive touchscreen, 16M colors',
    size: '3.5 inches (~52.0% screen-to-body ratio)',
    os: '',
  },
];

describe('<PhoneListContainerRaw/ >', () => {
  it('renders correctly with empty phones list', () => {
    const tree = renderer
      .create(withDefaultIntl(<PhoneListContainerRaw phones={[]} />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with phones list', () => {
    const tree = renderer
      .create(withDefaultIntl(<PhoneListContainerRaw phones={stubPhones} />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
