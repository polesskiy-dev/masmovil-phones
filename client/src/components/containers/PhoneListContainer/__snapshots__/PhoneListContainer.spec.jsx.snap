// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhoneListContainerRaw/ > renders correctly with empty phones list 1`] = `
<section
  className="phones-list"
/>
`;

exports[`<PhoneListContainerRaw/ > renders correctly with phones list 1`] = `
<section
  className="phones-list"
>
  <article
    className="phone-card"
  >
    <img
      alt="Apple iPhone"
      className="phone-card__image"
      src="https://upload.wikimedia.org/wikipedia/commons/0/02/IPhone_2G_PSD_Mock.png"
    />
    <div
      className="phone-card__description"
    >
      <div
        className="phone-card__text-description"
      >
        <p
          className="phone-card__name"
        >
          Apple iPhone
        </p>
        <ul
          className="phone-card__details"
        >
          <li>
            TFT capacitive touchscreen, 16M colors
          </li>
          <li>
            
          </li>
        </ul>
      </div>
      <button
        className="details-button"
        onClick={[Function]}
      >
        <span>
          Details
        </span>
      </button>
    </div>
  </article>
</section>
`;
